runhouse/rns/rns_client.py

```diff
@@ -96,7 +96,11 @@ def token(self):
 
     @property
     def username(self):
-        return self._configs.get("username", None)
+        return self._configs.username or self._configs.get("username", None)
```

Comment:
because it gets the username from the default cache... in case we update the username (and we update if for testing purposes), it does not saved in the rns_client default cache. 

I could add some logic that updated the default cache in case we update the username, but tbh it feels a bit of an overkill to support this just for the sake of testing. wdyt?

Reply:
yeah but when we first set the username using the config yaml, we set self._username... And then it is reaturned even if we set a new value for `RH_USERNAME`
![image](https://github.com/user-attachments/assets/d3c13f23-908f-470f-9b69-e60e1c13ca6a)